src/postgres/contrib/yb_auh/yb_auh.c

```diff
@@ -175,24 +176,27 @@ yb_auh_main(Datum main_arg) {
 
 static void pg_collect_samples(TimestampTz auh_sample_time, uint16 sample_size_procs)
 {
-  LWLockAcquire(ProcArrayLock, LW_SHARED);
   LWLockAcquire(auh_entry_array_lock, LW_EXCLUSIVE);
-  int		procCount = ProcGlobal->allProcCount;
-  float8 sample_rate = 0;
-  if(procCount != 0)
-    sample_rate = (float)Min(sample_size_procs, procCount)/procCount;
-  for (int i = 0; i < procCount; i++)
-  {
-    PGPROC *proc = &ProcGlobal->allProcs[i];
-    if (proc != NULL && proc->pid != 0 && (random() < sample_rate * MAX_RANDOM_VALUE)){
-      auh_entry_store(auh_sample_time, proc->top_level_request_id, 0,
-                      proc->wait_event_info, "", proc->top_level_node_id,
-                      proc->client_node_host, proc->client_node_port,
-                      proc->queryid, auh_sample_time, sample_rate);
+  PgProcAuhNode *nodes_head = pg_collect_samples_proc();
+  PgProcAuhNode *current = nodes_head;
+  while (current != NULL) {
+    PGProcAUHEntryList proc = current->data;
+    int procCount = proc.numprocs;
+    float8 sample_rate = 0;
+    if (procCount != 0) {
+        sample_rate = (float)Min(sample_size_procs, procCount) / procCount;
+    }
```

Comment:
This will be same for a single iteration of `pg_collect_samples ` correct? If yes, can we move this out of the loop?